include/libopencm3/sam/4l/memorymap.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2016 Maxim Sloyko <maxim@sloyko.com>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef SAM4L_MEMORYMAP_H
+#define SAM4L_MEMORYMAP_H
+
+#include <libopencm3/cm3/common.h>
+
+#define PERIPH_BASE			(0x40000000U)
+
+/* --- SAM4L peripheral bridge A -------------------------------------------- */
+#define PERIPH_PBA_BASE			PERIPH_BASE
+/* (PERIPH_PBA_BASE + 0x4000*0) RESERVED */
+#define I2SC_BASE			(PERIPH_PBA_BASE + 0x4000*1)
+#define SPI_BASE			(PERIPH_PBA_BASE + 0x4000*2)
+/* (PERIPH_PBA_BASE + 0x4000*3) RESERVED */
+#define TC0_BASE			(PERIPH_PBA_BASE + 0x4000*4)
+#define TC1_BASE			(PERIPH_PBA_BASE + 0x4000*5)
+#define TWIMS0_BASE			(PERIPH_PBA_BASE + 0x4000*6)
```

Comment:
please don't do this style.  I know it's related to how your m4 macros worked out, but it's too tedious for a reader to see what the address actually is when they're comparing with the reference manual.